src/autoscaling_alloc_buffer.jl

```diff
@@ -0,0 +1,121 @@
+const DEFAULT_NEW_BUFFER_SIZE = 2^27 # 128 MB
+const DEFAULT_HISTORY_SIZE = 5
+const DEFAULT_DESIRED_MAX_BUFFERS = 1
+
+mutable struct AutoscalingAllocBuffer
+    main_buffer::AllocBuffer{Vector{UInt8}}
+    const additional_buffers::Vector{AllocBuffer{Vector{UInt8}}}
+    new_buffer_size::Int
+    const main_buffer_size_history::Vector{Int}
```

Comment:
So I remember: change this to `total_used_history` and modify the realloc conditions so we actually can shrink main buffer size sometimes